src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -401,9 +403,33 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                                     const WORD scanCode,
                                     const ::Microsoft::Terminal::Core::ControlKeyStates modifiers)
     {
+        if (ch == L'\x3') // Ctrl+C or Ctrl+Break
+        {
+            _handleControlC();
+        }
+
         return _terminal->SendCharEvent(ch, scanCode, modifiers);
     }
 
+    void ControlCore::_handleControlC()
+    {
+        if (!_midiAudioSkipTimer)
+        {
+            _midiAudioSkipTimer = _dispatcher.CreateTimer();

```

Comment:
Is this the best way to create timers in Windows Terminal?

Reply:
The documentation isn't particularly great
> ## DispatcherQueueTimer.Start Method
> ### Definition
> Starts the timer.

However wading through the underlying code for this, the timer is canceled before it gets restarted.